### PR TITLE
KDESKTOP-1019-macOS-LiteSync-File-deleted-if-app-restarted-during-upload

### PR DESCRIPTION
**Pull request by ClementKunz**
Several changes have been implemented to avoid the Delete operations described in [KDESKTOP-1019](https://infomaniak.atlassian.net/browse/KDESKTOP-1019):

1. We do not delete dehydrated placeholders that the app tried to upload. Instead, we rename them appending the "blacklisted" suffix, so that they are excluded from sync.
2. We do not set a status to a file which is being uploaded. Therefore, in the _Finder_, the files will not have any statuses until the end of the upload.
3. ~~Finally, we reinforce the detection of dehydrated placeholders by also retrieving the number of memory blocks used that is the number `st_blocks` obtained from the call to `lstat`. Therefore, a placeholder is considered dehydrated only if this number is `0`.~~ Finally, the use of `st_blocks` was misleading for files with thumbnails. Indeed, dehydrated placeholder with a thumbnail has a `st_blocks` value higher than 0.

[KDESKTOP-1019]: https://infomaniak.atlassian.net/browse/KDESKTOP-1019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp [hunk 1]
```diff
@@ -80,11 +80,11 @@ void UploadSession::runJob() {
     }
     auto start = std::chrono::steady_clock::now();
 
-    if (_vfsForceStatus) {
-        if (!_vfsForceStatus(_filePath, true, 0, true)) {
-            LOGW_WARN(_logger, L"Error in vfsForceStatus - path=" << Path2WStr(_filePath).c_str());
-        }
-    }
+    // if (_vfsForceStatus) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/upload_session/uploadsession.cpp:83]: Are these comments intended?

File: src/libsyncengine/jobs/network/upload_session/uploadsessionstartjob.cpp [hunk 2]
```diff
@@ -37,11 +37,11 @@ UploadSessionStartJob::UploadSessionStartJob(int driveDbId, const NodeId &fileId
 }
 
 UploadSessionStartJob::~UploadSessionStartJob() {
-    if (_vfsForceStatus) {
-        if (!_vfsForceStatus(_filePath, true, 0, true)) {
-            LOGW_WARN(_logger, L"Error in vfsForceStatus for path=" << Path2WStr(_filePath).c_str());
-        }
-    }
+    // if (_vfsForceStatus) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/upload_session/uploadsessionstartjob.cpp:40]: Same question.

File: src/libsyncengine/jobs/network/uploadjob.cpp [hunk 3]
```diff
@@ -87,11 +87,11 @@ bool UploadJob::canRun() {
 }
 
 void UploadJob::runJob() noexcept {
-    if (_vfsForceStatus) {
-        if (!_vfsForceStatus(_filePath, true, 0, true)) {
-            LOGW_WARN(_logger, L"Error in vfsForceStatus - path=" << Path2WStr(_filePath).c_str());
-        }
-    }
+    // if (_vfsForceStatus) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/uploadjob.cpp:90]: Idem.

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 4]
```diff
@@ -286,11 +286,14 @@ void ExecutorWorker::handleCreateOp(SyncOpPtr syncOp, std::shared_ptr<AbstractJo
         }
 
         if (isDehydratedPlaceholder) {
-            // Delete the file
-            LocalDeleteJob deleteJob(_syncPal->driveDbId(), _syncPal->_localPath, relativeLocalFilePath, true,
-                                     syncOp->affectedNode()->id().has_value() ? syncOp->affectedNode()->id().value() : "");
-            deleteJob.setBypassCheck(true);
-            deleteJob.runSynchronously();
+            // // Delete the file
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/propagation/executor/executorworker.cpp:289]: To be removed?

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 5]
```diff
@@ -865,11 +868,11 @@ bool ExecutorWorker::convertToPlaceholder(const SyncPath &relativeLocalPath, boo
         return false;
     }
 
-    bool isSyncing = hydrated;
-    if (!_syncPal->vfsForceStatus(absoluteLocalFilePath, isSyncing, 100, hydrated)) {
-        LOGW_WARN(_logger, L"Error in vfsForceStatus: " << Utility::formatSyncPath(absoluteLocalFilePath).c_str());
-        return false;
-    }
+    // bool isSyncing = hydrated;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/propagation/executor/executorworker.cpp:871]: Removed?

File: test/server/vfs/mac/testlitesyncextconnector.cpp [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        CPPUNIT_ASSERT(!isHydrated);
+    }
+    // Get status of an existing file that is not yet a placeholder
+    {
+        bool isPlaceholder = false;
+        bool isHydrated = false;
+        bool isSyncing = false;
+        int progress = 0;
+
+        // vfsGetStatus returns `true` if the file path indicates a non-existing item.
```
Comment (luc-guyot-infomaniak) [inline test/server/vfs/mac/testlitesyncextconnector.cpp:57]: So, this is the same test as above?

File: test/server/vfs/mac/testlitesyncextconnector.cpp [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            LiteSyncExtConnector::vfsGetStatus("/Users/clement/Projects/desktop-kDrive/test/test_ci/test_pictures/picture-1.jpg",
+                                               isPlaceholder, isHydrated, isSyncing, progress, logger));
+        CPPUNIT_ASSERT(!isPlaceholder);
+        CPPUNIT_ASSERT(!isHydrated);
+    }
+    {
+        FileStat buf;
+        IoError ioError = IoErrorUnknown;
+        IoHelper::getFileStat("/Users/clement/kDrive2/Move-Delete_tree.pdf", &buf, ioError);
+        int64_t testSize = buf.size;
```
Comment (luc-guyot-infomaniak) [inline test/server/vfs/mac/testlitesyncextconnector.cpp:69]: To be continued?

File: src/server/vfs/mac/CMakeLists.txt [hunk 8]
```diff
@@ -2,56 +2,56 @@ find_package(Qt6)
 
 # Target
 add_library("${libsyncengine_NAME}_vfs_mac" SHARED
-    vfs_mac.h vfs_mac.cpp
-    litesyncextconnector.mm
-    ../../../libcommonserver/vfs.h ../../../libcommonserver/vfs.cpp
-    ../../../libcommonserver/plugin.h ../../../libcommonserver/plugin.cpp
-    ../../../common/filepermissionholder.h ../../../common/filepermissionholder.cpp
-    ../../../common/filesystembase.h ../../../common/filesystembase.cpp
+        vfs_mac.h vfs_mac.cpp
```
Comment (luc-guyot-infomaniak) [inline src/server/vfs/mac/CMakeLists.txt:5]: Your IDE is formatting `CMakeList.txt` files. That's fine if the output looks better. But we could do all changes of this sort in a separate PR.

Comment (ClementKunz) [reply src/server/vfs/mac/CMakeLists.txt:5]: I have deactivated the auto formatting on CMake files

File: test/server/vfs/mac/testlitesyncextconnector.cpp (hunk 7 above)
Comment (ClementKunz) [reply test/server/vfs/mac/testlitesyncextconnector.cpp]: Yes, this file was a draft that I wanted to use to detect the problem. Should never have been pushed, at least not like that.
I have removed it.

File: test/server/vfs/mac/testlitesyncextconnector.cpp (hunk 6 above)
Comment (ClementKunz) [reply test/server/vfs/mac/testlitesyncextconnector.cpp]: Same as below

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp (hunk 1 above)
Comment (ClementKunz) [reply src/libsyncengine/jobs/network/upload_session/uploadsession.cpp]: I have removed them everywhere

**Review by luc-guyot-infomaniak — DISMISSED** [review]

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**